### PR TITLE
[move-prover] include the experimental code in df for verification

### PR DESCRIPTION
**Pull request by mengxu-fb**
Verification of diem-framework experimental code is turned-off
in #9718 due to symbolic links not being properly handled. This is
wrong and in fact, has caused some specifications to be violated.

This commit re-worked the move prover testsuite to include the
experimental code for verification.

The offending module is the newly refactored `DiemAccount.move` in
the experimental code. I'll leave it to the refactoring team
(@tzakian @emmazzz) to double check. The CI should fail with the
verification error.

To reproduce the result, pull down this PR locally and run
`cargo test DiemAccount.move` in `language/move-prover/tests`.
The `DiemAccount.move` in `diem_core` passes and the version in
`diem_exp` fails.

## Motivation

Cover the diem-framework experimental folder for verification

### Have you read the [Contributing Guidelines on pull requests](https://github.com/diem/diem/blob/main/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

CI

### CODE REVIEW

File: language/move-prover/tests/testsuite.rs [hunk 1]
```diff
@@ -251,14 +259,28 @@ fn collect_enabled_tests(reqs: &mut Vec<Requirements>, group: &str, feature: &Fe
...
+        .follow_links(false)
         .min_depth(1)
         .into_iter()
         .flatten()
     {
         if !entry.file_name().to_string_lossy().ends_with(".move") {
             continue;
         }
+
+        // TODO: this is to handle the awkwardness in the experimental folder in diem-framework
```
Comment (wrwg) [inline language/move-prover/tests/testsuite.rs:271]: Packages will be used soon, it would be good to land this soon as I'm going to refactor this.

**Review by wrwg — APPROVED** [review]

File: language/move-prover/tests/testsuite.rs (hunk 1 above)
Comment (mengxu-fb) [reply language/move-prover/tests/testsuite.rs:271]: Yes, package should be the solution.

The only thing that blocks it from landing is that the new `DiemAccount.move` in the experimental dir does not pass the verification... I could temporarily turn it off and the refactoring team can debug it later.

**Review by tzakian — APPROVED** [review]

**Comment by mengxu-fb** [comment]
Test failure at tear down step... try again

/land

**Comment by mengxu-fb** [comment]
failure of the CI is due to an error where the experimental df release is not updated for docs and built artifacts... now needs a re-stamping... @wrwg @tzakian

**Review by wrwg — APPROVED** [review]

**Comment by mengxu-fb** [comment]
Now this is faucet issue..., try again

/land

**Pull request closed without merging**